### DISCUSSION TITLE
명확하지만 너무 긴 메서드명 vs 명확하지 않지만 짧은 메서드명 중 어떤 것을 골라야할까요?

### DISCUSSION THREAD

**Discussion by gyungchan-jo** [2024-woowacourse-study/level-interview#106] [Clean Code]
.

**Comment by kunsanglee** [comment]
명확하지만 긴 메서드명을 선택하는 것이 더 바람직할 수 있습니다. 그 이유는 메서드가 수행하는 기능이나 역할을 명확하게 표현함으로써, 코드를 처음 접하는 다른 개발자들이 그 의미를 쉽게 이해할 수 있기 때문입니다. 이는 코드의 가독성과 유지보수성을 크게 향상시킵니다.

그러나 메서드명이 길어진다는 것은 해당 메서드가 여러 가지 역할을 수행하고 있을 가능성을 시사합니다. 이런 경우, 메서드 내에 다양한 관심사가 혼재되어 있을 수 있으므로, 메서드를 더 작은 단위로 분리하는 것이 좋습니다. 이를 통해 각 메서드가 단일 책임을 가지도록 하여, 더 명확하고 간결한 메서드명을 사용할 수 있게 됩니다.

따라서, 메서드명이 길어지는 것을 발견했을 때는, 메서드가 여러 작업을 수행하고 있는지 재검토해보고, 가능하다면 메서드를 더 작은 단위로 분리하여 각각의 메서드가 명확한 역할과 책임을 갖도록 하는 것이 중요합니다.

**Comment by seunghye218** [comment]
일반적으로 명확하고 설명이 포함된 메소드 이름을 선호합니다. 왜냐하면 좋은 가독성과 유지 관리성으로 인한 이점이 긴 메서드명으로 인한 단점보다 더 크다고 생각하기 때문입니다. 그러나 길지만 더 설명적인 이름이 필요하면 함수를 분리해야 하는 경우라고 생각합니다.

그래도 중요한 점은 특정 프로젝트와 개발팀에 적절한 컨벤션을 사용하는 것입니다.
